### PR TITLE
feat: Add signup screen.

### PR DESCRIPTION
**Pull request by BURG3R5**

### CODE REVIEW

File: lib/presentation/signup/bloc/sign_up_bloc.dart [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+      instituteController: TextEditingController(),
+      nameController: TextEditingController(),
+      passwordController: TextEditingController(),
+      usernameController: TextEditingController(),
+      handleControllers: {
+        'codechef': TextEditingController(),
+        'hackerrank': TextEditingController(),
+        'codeforces': TextEditingController(),
+        'spoj': TextEditingController(),
+      },
```
Comment (praeclarumjj3) [inline lib/presentation/signup/bloc/sign_up_bloc.dart:57]: No LeetCode while asking for the platform handles? 

CC: @akanksha-v

File: lib/presentation/signup/verify_screen.dart [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+              Text(
+                AppStrings.signUpSuccess,
+                style: AppStyles.h5.copyWith(color: AppColors.grey3),
+              ),
+              const Spacer(flex: 2),
+              Row(
+                mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                children: <Widget>[
+                  resendEmailButton(userId),
+                  loginButton(username, password, profilePicture),
```
Comment (praeclarumjj3) [inline lib/presentation/signup/verify_screen.dart:55]: Fix the overflow here as mentioned on slack.

**Review by praeclarumjj3 — CHANGES_REQUESTED** [review]

File: lib/presentation/signup/bloc/sign_up_bloc.dart (hunk 1 above)
Comment (BURG3R5) [reply lib/presentation/signup/bloc/sign_up_bloc.dart:58]: Fixed in bc9a94c.

File: lib/presentation/signup/verify_screen.dart (hunk 2 above)
Comment (BURG3R5) [reply lib/presentation/signup/verify_screen.dart:55]: Fixed in 9c87592.

**Review by praeclarumjj3 — APPROVED** [review]

**Pull request merged**